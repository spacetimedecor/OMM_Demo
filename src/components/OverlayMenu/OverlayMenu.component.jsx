import StyledOverlayMenu from "./OverlayMenu.styled";
import {StyledOverlayMenuItem} from "./OverlayMenu.styled";
import {default as React} from 'react';
import {
  useLocation
} from "react-router-dom";

export const OverlayMenu = (props) => {
  return (
    <StyledOverlayMenu {...props}>
      {props.children}
    </StyledOverlayMenu>
  );
}

export const OverlayMenuItem = (props) => {
  const {label, to, ...rest} = props;
  const location = useLocation();
  const selected = to === location.pathname;

  return (
    <StyledOverlayMenuItem
      to={to}
      className={selected && 'selected'}
      {...rest}
    >
      {label}
    </StyledOverlayMenuItem>
  )
}

export default OverlayMenu;
