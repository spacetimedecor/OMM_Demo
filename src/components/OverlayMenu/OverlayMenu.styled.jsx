import styled from "styled-components";
import { header1 } from "../common";
import { Link } from "react-router-dom";
import { breakpoints } from "../../utils/breakpoints";

const StyledOverlayMenu = styled("span")`
  color: ${({ theme }) => theme.colors.primaryColor};
  width: 100vw;
  height: 100vh;
  text-align: center;
  position: fixed;
  background-color: ${({ backgroundColor, theme }) => backgroundColor || theme.overlayMenu.backgroundColor};;
  display: flex;
  justify-content: center;

  @media screen and (max-width: ${breakpoints.tablet}) {
    flex-direction: column;
  }
  
  ${ ({ onClick }) => onClick ? '&:hover { cursor: pointer }' : '' }
  
  > .selected {
    color: white;
  }
`;

export const StyledOverlayMenuItem = styled(Link)`

  width: min-content;
  white-space: nowrap;
  justify-self: center;
  align-self: center;
  
  padding: 15px 50px 15px 50px;
  
  ${header1};

  @media screen and (max-width: ${breakpoints.tablet}) {
    border-left: none;
  }
  
  border-left: white solid 1px;
  color: #868686;
  transition: color 0.25s;
  
  &:hover {
    color: #e66465;
  }

  &:first-of-type {
    border-left: none;
  }
`;

export default StyledOverlayMenu;
