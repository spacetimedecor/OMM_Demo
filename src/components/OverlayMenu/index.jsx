export { default } from "./OverlayMenu.component";
