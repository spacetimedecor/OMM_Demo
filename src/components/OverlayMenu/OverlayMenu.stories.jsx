import {OverlayMenu, OverlayMenuItem} from "./OverlayMenu.component";

import React from "react";

export default {
    title: 'Components/OverlayMenu',
    component: OverlayMenu,
    parameters: {
        layout: 'fullscreen',
    },
    argTypes: {
        backgroundColor: { control: 'color' },
    }
};

const Template = (args) => <OverlayMenu {...args}>
    <OverlayMenuItem label="HOME" to="/" />
    <OverlayMenuItem label="CREATIVE" to="/creative" />
    <OverlayMenuItem label="TECHNOLOGY" to="/technology" />
</OverlayMenu>;

export const Primary = Template.bind({});
Primary.args = {
    primary: true,
};
