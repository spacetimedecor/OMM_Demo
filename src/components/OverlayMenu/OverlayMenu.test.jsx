import * as React from "react";

import themed from "../../theme/themed";
import OverlayMenu from "./OverlayMenu.component";

describe("OverlayMenu", () => {
  it("should render correctly", () => {
    const component = themed(
      <OverlayMenu>
        Test
      </OverlayMenu>,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
