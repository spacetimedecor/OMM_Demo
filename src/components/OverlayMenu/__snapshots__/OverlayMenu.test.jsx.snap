// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayMenu should render correctly 1`] = `
<span
  className="sc-gsDKAQ bVwgo"
>
  Test
</span>
`;
